code/test/general.js

```diff
@@ -0,0 +1,2306 @@
+    assert.notEqual(nonce, aliceOldState.nonce, 'alice wallet.nonce should have been changed')
+    assert.equal(nonce.toNumber(), aliceOldState.nonce + 1, 'alice wallet.nonce should have been changed')
+    aliceOldState.nonce = nonce.toNumber()
+    // lastOperationTime
+    let lastOperationTime = await alice.wallet.lastOperationTime()
+    assert.notStrictEqual(lastOperationTime, aliceOldState.lastOperationTime, 'alice wallet.lastOperationTime should have been updated')
+    aliceOldState.lastOperationTime = lastOperationTime.toNumber()
+    // recoveryAddress
+    let aliceGetInfo = await alice.wallet.getInfo()
+    assert.notDeepEqual(aliceGetInfo, aliceOldState.info, 'alice wallet.getInfo recoveryAddress should have been changed')
+    assert.equal(aliceGetInfo[6].toString(), carol.wallet.address, 'alice wallet.getInfo recoveryAddress should equal carol.wallet.address')
+    aliceOldState.info.recoveryAddress = aliceGetInfo[6].toString()
+    // commits
+    let allCommits = await alice.wallet.getAllCommits()
+    assert.notDeepEqual(allCommits, aliceOldState.allCommits, 'alice wallet.allCommits should have been updated')
+    aliceOldState.allCommits = allCommits
+    // check alice
+    await TestUtil.checkONEWalletStateChange(aliceOldState, aliceCurrentState)
+  })
+
+  // ==== RECOVER =====
+  // Test recover all funds and tokens from alices wallet
+  // Expected result: will be transferred tos her last resort address (currently bob's last resort address)
+  it('OPERATION 6 RECOVER: must be able to recover assets', async () => {
+    // create wallets and token contracts used througout the tests
+    let alice = await TestUtil.makeWallet('TG-OP6-1', accounts[0], EFFECTIVE_TIME, DURATION)
+    let aliceOldState = await TestUtil.getONEWalletState(alice.wallet)
+    let aliceCurrentState = await TestUtil.getONEWalletState(alice.wallet)
+    const { testerc20 } = await TestUtil.makeTokens(accounts[0])
+    let aliceBalanceERC20
+    let aliceWalletBalanceERC20
+    // transfer ERC20 tokens from accounts[0] (which owns the tokens) to alices wallet
+    await testerc20.transfer(alice.wallet.address, 1000, { from: accounts[0] })
+    aliceCurrentState = await TestUtil.getONEWalletState(alice.wallet)
+    aliceBalanceERC20 = await testerc20.balanceOf(alice.wallet.address)
+    assert.equal(1000, aliceBalanceERC20, 'Transfer of 1000 ERC20 tokens to alice.wallet succesful')
+    aliceWalletBalanceERC20 = await alice.wallet.getBalance(ONEConstants.TokenType.ERC20, testerc20.address, 0)
+    assert.equal(1000, aliceWalletBalanceERC20, 'Transfer of 1000 ERC20 tokens to alice.wallet checked via wallet succesful')
+    console.log('Funded ERC20')
+
+    // Begin Tests
+    let testTime = Date.now()
+
+    testTime = await TestUtil.bumpTestTime(testTime, 60)
+    // create randomSeed
+    let randomSeed = new Uint8Array(new BigUint64Array([0n, BigInt(testTime)]).buffer)
+    // recover Alices wallet
+    //
+    await TestUtil.transactionExecute(
```

Comment:
see basic.js:150. Recovery has special args. Probably shouldn't use this generation execution function